tests/unit/amo/pages/TestGuides.js

```diff
@@ -69,9 +70,59 @@ describe(__filename, () => {
       dispatchSpy,
       fetchGuidesAddons({
         guids,
-        errorHandlerId: root.instance().props.errorHandler.id,
+        errorHandlerId: errorHandler.id,
+      }),
+    );
+
+    sinon.assert.calledOnce(dispatchSpy);
+  });
+
+  it('does not fetch guides addons while loading', () => {
+    const { store } = dispatchClientMetadata();
+    const slug = 'privacy';
+    const errorHandler = createStubErrorHandler();
+    const sections = getSections(slug, fakeI18n());
+    const guids = sections.map((section) => section.addonGuid);
+
+    // This simulates the initial fetch for addons.
+    store.dispatch(
+      fetchGuidesAddons({
+        guids,
+        errorHandlerId: errorHandler.id,
+      }),
+    );
+
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    render({ store });
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it('does not fetch guides addons if addons has already been set', () => {
+    const { store } = dispatchClientMetadata();
+    const errorHandler = createStubErrorHandler();
+    const guid = 'test';
+    const addons = {
+      [guid]: {},
+    };
+
+    // This simulates the initial fetch for addons.
+    store.dispatch(
+      fetchGuidesAddons({
+        guids: [guid],
+        errorHandlerId: errorHandler.id,
       }),
     );
+
+    // This simulates loading addons which updates the loading state.
+    _loadAddonResults(store);
```

Comment:
You should probably use:

```js
store.dispatch(loadAddonResults({  addons });
```

and not inject `addons` in `render()`

Reply:
I think you've already fixed this but yes I learned that we should not inject mapStateToProps vars here ,right? Still got to learn a bit more about testing:)

Reply:
Yep, we should only inject props declared in `Props`, not from `InternalProps`. Unfortunately we still have a few test cases like this...

Reply:
The rationale for this is that when we inject a mapped prop in a test, we are bypassing all of our mapping logic and moving away from real life usage. If we dispatch the Redux action instead, the test will cover the mapping logic which ensures that the component is working correctly. Additionally, dispatching the Redux action will keep the test in sync with the state and reducers which is what we want.